controller/src/main/java/io/pravega/controller/task/Stream/TxnSweeper.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.task.Stream;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.concurrent.FutureHelpers;
+import io.pravega.controller.store.stream.StreamMetadataStore;
+import io.pravega.controller.store.stream.TxnStatus;
+import io.pravega.controller.store.task.TxnResource;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Duration;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * Sweeper for transactions orphaned by failed controller processes.
+ */
+@Slf4j
+public class TxnSweeper {
```

Comment:
I am planning to add `pre-condition` and `post-condition` for `CreateTxn`, `PingTxn`, `CommitTxn`, `AbortTxn` methods and event processor's `process` methods, along with valid invariants. I will do this in one commit. I will mention the commit id in a following comment so that it is easy to validate the invariants.